Pion fails to build on linux: BOOST_NOEXCEPT does not name a type

Trying to build pion on ElementaryOS Luna fails with following error:

> Making all in src
> make[1]: Entering directory `/home/jsokolow/Test/pion-develop/src'
>  CXX    spdy_parser.lo
> In file included from spdy_parser.cpp:15:0:
> ../include/pion/spdy/parser.hpp:55:28: error: expected ';' at end of member declaration
> ../include/pion/spdy/parser.hpp:55:34: error: 'BOOST_NOEXCEPT' does not name a type
> make[1]: *** [spdy_parser.lo] Error 1

Boost version i have installed is 1.48
autogen and config gave no errors. I don't have cmake 2.8.10 installed to test whether cmake would get it working.

--- a/include/pion/spdy/parser.hpp
+++ b/include/pion/spdy/parser.hpp
@@ -21,7 +21,11 @@
 #include <pion/spdy/decompressor.hpp>
 
 #ifndef BOOST_SYSTEM_NOEXCEPT
-    #define BOOST_SYSTEM_NOEXCEPT BOOST_NOEXCEPT
+    #ifndef BOOST_NOEXCEPT
+        #define BOOST_SYSTEM_NOEXCEPT
+    #else
+        #define BOOST_SYSTEM_NOEXCEPT BOOST_NOEXCEPT
+    #endif
 #endif
 
 

